src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/CommandParsers.cs

```diff
@@ -11,11 +11,14 @@
 
 namespace NuGet.CommandLine.XPlat.Commands
 {
+    /// <summary>Generated with CommandParsers.tt</summary>
+    [System.CodeDom.Compiler.GeneratedCode("Type:Microsoft.VisualStudio.TextTemplating.VSHost.TextTemplatingService; Assembly:Microsoft.VisualStudio.TextTemplating.VSHost, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "AssemblyFileVersion:17.0.33201.384")]
```

Comment:
When the version of VS changes, I don't want `git diff` to show changes in this file when I haven't made any changes to the `.tt` file. I really don't think that putting the assembly version in a file that's checked in is a good idea.

Reply:
This also happens when you update .resx generated code. Sorry. I could not find any other efficient way of doing this.

Reply:
Also, I expect changes on these files around every VS major version

Reply:
`.designer.cs` files don't get regenerated every time you build, so they don't change unless you actually modify the `.resx` files.  Doesn't the `.tt` files get re-run every build (or at least first build after project load)?

What happens if you re-open the solution in public preview or GA version of VS, and build? Do the `.cs` files change despite the `.tt` files note changing?

Reply:
> Also, I expect changes on these files around every VS major version

What about if an external contributor is on linux and therefore can't use VS, so they use `dotnet-tt` instead? Won't that change the tool name, and therefore once checked in, every time anyone opens the solution in VS, it again shows changes?  I think we also can't use the host name for the same reason (unless we gitignore the generated files, like what Donnie did in one of the VS assemblies (I think the PM UI assembly. It's easy to spot because VS adds a warning in the error list on solution open)

Reply:
To avoid confusion, I removed assembly information in comments. Here are the answers to your questions.

> .designer.cs files don't get regenerated every time you build, so they don't change unless you actually modify the .resx files. Doesn't the .tt files get re-run every build (or at least first build after project load)?

Those .tt files are not generated everytime you build, because nuget.commandline.xplat.csproj project file does not have the TransformOnBuild MSBuild property set to true.


> What happens if you re-open the solution in public preview or GA version of VS, and build? Do the .cs files change despite the .tt files note changing?

If shipped assembly file versions are changed, and, if you run the T4 templates, then, it would add the updated assembly information